### PR TITLE
fix: allow ip hosts and hostnames

### PR DESCRIPTION
**Pull request by magnasilvar**
## TL;DR

VS Code users at my company are facing a bug that at first sight seemed to be linked to the IDE (because our unit tests run from the command line), but which in reality is related to a `pytest-socket` bug (highlighted by the creation of a socket on the hostname `localhost` by the [vscode-python](https://github.com/microsoft/vscode-python) extension).

---

## Problem

Our following code from `conftest.py` makes it impossible to use the **Testing** tab in **VS Code** (no matter which unit test is run).

```python
# ...
from pytest_socket import socket_allow_hosts
# ...

# ...
def pytest_runtest_setup():
    socket_allow_hosts(
        allowed=["localhost", "127.0.0.1", "::1"], allow_unix_socket=True
    )
# ...
```

Error observed:
```
Error attempting to connect to extension communication socket[vscode-pytest]: A test tried to use socket.socket.connect() with host "localhost" (allowed: "localhost,127.0.0.1,::1").
```

---

## Why?

The error message from `pytest-socket` is wrong:
* `localhost` is not compared to `localhost`,`127.0.0.1` and `::1`.
* But to `127.0.0.1`,`127.0.0.1` and `::1`.

This is because the allowed hosts used for comparison are previously converted to IP addresses. Therefore, a comparison with a host that is not an IP address can never succeed.

---

## Affected versions

`pytest-socket` version | Bug presence
--------|--------------
`0.6.0` | Yes
`0.5.1` | No

---

### VS Code info

```
Version: 1.84.2 (Universal)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:52:33.687Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.1.0
```

### Extension version
```
Name: Python
Id: ms-python.python
Version: 2023.20.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.python
```

---

## Related issues / PR

This PR resolves:
* #246 
* #203 
* https://github.com/microsoft/vscode-python/issues/22383

### CODE REVIEW

**Comment by magnasilvar** [comment]
Hi @miketheman 👋 Can you take a look when you have time? 🙏

**Comment by magnasilvar** [comment]
Happy New Year @miketheman 🎉
From the reactions on this PR, it seems that many people have the same issue, is it possible for you to review it? 🙏

**Review by miketheman — APPROVED** [review]

**Comment by miketheman** [comment]
Thanks @magnasilvar  - I'll likely cut a fresh release in the next few days.
